react/features/local-recording/components/LocalRecordingInfoDialog.js

```diff
@@ -0,0 +1,332 @@
+/* @flow */
+
+import moment from 'moment';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { translate } from '../../base/i18n';
+import {
+    PARTICIPANT_ROLE,
+    getLocalParticipant
+} from '../../base/participants';
+
+import { statsUpdate } from '../actions';
+import { recordingController } from '../controller';
+
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link LocalRecordingInfoDialog}.
+ */
+type Props = {
+
+    /**
+     * Redux store dispatch function.
+     */
+    dispatch: Dispatch<*>,
+
+    /**
+     * Current encoding format.
+     */
+    encodingFormat: string,
+
+    /**
+     * Whether the local user is the moderator.
+     */
+    isModerator: boolean,
+
+    /**
+     * Whether local recording is engaged.
+     */
+    isOn: boolean,
+
+    /**
+     * The start time of the current local recording session.
+     * Used to calculate the duration of recording.
+     */
+    recordingStartedAt: Date,
+
+    /**
+     * Stats of all the participant.
+     */
+    stats: Object,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function
+}
+
+/**
+ * The type of the React {@code Component} state of
+ * {@link LocalRecordingInfoDialog}.
+ */
+type State = {
+
+    /**
+     * The recording duration string to be displayed on the UI.
+     */
+    durationString: string
+}
+
+/**
+ * A React Component with the contents for a dialog that shows information about
+ * local recording. For users with moderator rights, this is also the "control
+ * panel" for starting/stopping local recording on all clients.
+ *
+ * @extends Component
+ */
+class LocalRecordingInfoDialog extends Component<Props, State> {
+
+    /**
+     * Saves a handle to the timer for UI updates,
+     * so that it can be cancelled when the component unmounts.
+     */
+    _timer: ?IntervalID;
+
+    /**
+     * Constructor.
+     */
+    constructor() {
+        super();
```

Comment:
React component constructors take in props and props should be passed into the super call.